### ISSUE TITLE
Package: syntax error, unexpected '->' (T_OBJECT_OPERATOR), expecting ',' or ')'

### ISSUE THREAD

**Issue by phpjob** [symfony/symfony#29475]
**Symfony version(s) affected**: 4.2.0

**Description**  
Since i updated following package "[symfony/dependency-injection](https://github.com/symfony/dependency-injection)" (4.1.8 -> 4.2.0), i got the following Error: 

```
Package: syntax error, unexpected '->' (T_OBJECT_OPERATOR), expecting ',' or ')'
```
Somewhere in a auto generated Dependency File

```
<?php

use Symfony\Component\DependencyInjection\Argument\RewindableGenerator;
use Symfony\Component\DependencyInjection\Exception\RuntimeException;

// This file has been auto-generated by the Symfony Dependency Injection Component for internal use.
// Returns the public 'guzzle.client.coupies_api_client' shared service.

$a = \GuzzleHttp\HandlerStack::create();
$a->push(new \App\Guzzle\Mapudo\Middleware\Rewind()->rewind());
$a->push($this->load('getGuzzleBundle_Middleware_Log_CoupiesApiClientService.php')->log());
$a->unshift($this->load('getGuzzleBundle_Middleware_EventDispatch_CoupiesApiClientService.php')->dispatch());

return $this->services['guzzle.client.coupies_api_client'] = new \GuzzleHttp\Client(array('base_uri' => NULL, 'handler' => $a, 'headers' => array('User-Agent' => 'Firefox')));
```

there is a syntax error, exactly:

```
$a->push(new \App\Guzzle\Mapudo\Middleware\Rewind()->rewind());
```

Must be:

```
$a->push((new \App\Guzzle\Mapudo\Middleware\Rewind())->rewind());
```

I tested it back to v4.2.0-BETA1, but it still exists.

I tried, started on version v4.2.0, to revert following class [PhpDumper](https://github.com/symfony/dependency-injection/blob/v4.2.0/Dumper/PhpDumper.php) until 4.1.8 and it looks, like there in the class is somewhere the error.

Maybe you know already the error @nicolas-grekas ?

**How to reproduce**  
I will give you more information, if you need it. But since the system is complex, i will try to get a day to debug.

**Additional context**  
```
Package: syntax error, unexpected '->' (T_OBJECT_OPERATOR), expecting ',' or ')'
```

**Comment by nicolas-grekas** [comment]
This should be fixed in 4.2@dev

**Comment by phpjob** [comment]
Tested in version v4.2.1 - issue fixed. Thanks a lot! 👍
